feature/lessons-widget-scaling WIP

free2fqn opened the pull request:
Ребята, я в тупике. Пытаюсь сделать сервис, который собирает JSON, отражающий структуру глав/степов/уроков на платформе. Вложенность ограниченная (делаю вручную ибо тупой и с обходом не вывез).
Не бейте по рукам за использование вывода в консоль и мейна, благодаря им я понимаю, что моя функция точно знает о существовании уроков (т.е. о 1.html и 13.html):
![image](https://user-images.githubusercontent.com/22982519/155098716-623d9231-363c-4659-a24f-9fdd949dcb98.png)
Но на постман приходит такая лабуда, уже без уроков:
![image](https://user-images.githubusercontent.com/22982519/155099289-d68c7fcc-bf56-4c97-ab76-d3150f273f89.png)
Прошу помощи, что я проглядел и почему у меня "обрезаются" списки уроков?

free2fqn committed:
refactored structure of lessons.
added endpoint "/structureOf/{course}".
Created LessonStructureDTO, LessonStructureException, LessonStructureService, LessonStructureServiceImpl.
Temporary using main() in PlatformOrchestratorApplication.

nekromant322 commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureServiceImpl.java:
```diff
@@ -0,0 +1,63 @@
+        List<String> listOfChapters;
+        List<String> listOfSteps;
+        List<String> listOfLessons;
+        String stringPathToCourse = "C:\\Users\\Schneider\\IdeaProjects\\platform\\platform_orchestrator\\src\\main\\resources\\templates\\lessons" + "\\" + courseName;
```
-> относительный путь?

nekromant322 commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureServiceImpl.java:
```diff
@@ -0,0 +1,63 @@
+            structureMap.put(chapter, stepLessonStructure);
+        }
+        lessonStructureDTO.setStructureMap(structureMap);
+        return lessonStructureDTO; //почему возвращается без уроков? в консоли уроки выводятся
```
понятия не имею почему, ты же разработчик, а не я, продебажь

nekromant322 commented on platform_orchestrator/src/main/java/com/override/exception/LessonStructureException.java:
```diff
@@ -0,0 +1,6 @@
+package com.override.exception;
+
+//Правильно понял логику, что BaseOrchestratorException для этого как раз и нужен?
+public class LessonStructureException extends BaseOrchestratorException{
```
ну, если будет вылетать этот эксепшон, то он будет отрабатываться эдвайсом, статус без явного указания INTERNAL_SERVER_ERROR буит

nekromant322 commented on platform_orchestrator/src/main/java/com/override/controller/LessonController.java:
```diff
@@ -1,16 +1,42 @@
+        try {
+            return lessonStructureService.scanLessonStructure(course);
+        } catch (IOException e) {
+            // почему здесь нужен try-catch? все IOException ловятся уже в lessonStructureService.getDirectoryStructure()
```
вроде не нужен, там throws нет нигде

free2fqn committed:
removed using LessonStructureDTO, now using JSONObject.
added suppor of org.json at Orchestrator's pom.xml.
got a problem: json table of contents is unordered, solve it later

free2fqn committed:
removed using JSONObject, now using JsonObject.
added supporе of org.google.gson at Orchestrator's pom.xml.
now structure of course-folder sends good.

free2fqn committed:
WIP
added Sidebar-fragment at 1.html

free2fqn committed:
WIP
added Sidebar-fragment at all lesson-files.
Refactored lesson-files at columns.
Removed LessonStructureDTO.
Clean up.

free2fqn committed:
WIP
added Sidebar-fragment at all lesson-files.
Refactored lesson-files at columns.
Removed LessonStructureDTO.
Clean up.

free2fqn commented:
проблему обозначил TODO в последнем коммите

kemenchik commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureServiceImpl.java:
```diff
@@ -0,0 +1,65 @@
+
+    @Override
+    public JsonObject scanLessonStructure(String courseName) {
+        String stringPathToCourse = "C:\\Users\\Schneider\\IdeaProjects\\platform\\platform_orchestrator\\src\\main\\resources\\templates\\lessons" + "\\" + courseName;
```
Полный путь все еще

kemenchik commented on platform_orchestrator/src/main/java/com/override/PlatformOrchestratorApplication.java:
```diff
@@ -1,17 +1,35 @@
 
     public static void main(String[] args) {
-        SpringApplication.run(PlatformOrchestratorApplication.class, args);
+        ApplicationContext context = SpringApplication.run(PlatformOrchestratorApplication.class, args);
+//        List<String> stringList = new LinkedList<>(Arrays.asList("first", "second", "third", "fourth"));
+//        ListIterator<String> stringListIterator = stringList.listIterator(stringList.size());
+//        while (stringListIterator.hasPrevious()) {
+//            System.out.println(stringListIterator.previous());
+//        }
+//        System.out.println("_____________");
+//        stringListIterator = stringList.listIterator();
+//        while (stringListIterator.hasNext()) {
+//            System.out.println(stringListIterator.next());
+//        }
```
Че это вообще

kemenchik commented on platform_orchestrator/src/main/resources/static/js/lessonSidebar.js:
```diff
@@ -0,0 +1,62 @@
+/*TODO как в javascript сделать сюда инжект названия курса? Вообще идей нет, как считывать название выбранного  курса */
+window.onload = init("core");
```
Из window.href по урлу получать название курса

kemenchik commented on platform_orchestrator/src/main/resources/templates/lessons/core/introduction/step1/2.html:
```diff
@@ -0,0 +1,37 @@
+<!DOCTYPE html>
+<html lang="en">
```
Нет импорта thymleaf, мб работать не будет

free2fqn replied:
работает

free2fqn committed:
Merge remote-tracking branch 'origin/feature/lessons-widget-scaling' into feature/lessons-widget-scaling

free2fqn committed:
now lesson Controller caches structure of all courses in HashMap.

free2fqn committed:
fixed absolute links

free2fqn committed:
privated and renamed postConstruct in LessonStructureServiceImpl

free2fqn committed:
clean up

free2fqn committed:
clean up

nekromant322 commented on platform_orchestrator/src/main/java/com/override/controller/LessonController.java:
```diff
@@ -1,16 +1,30 @@
 
+    private final LessonStructureService lessonStructureService;
+
+    @Autowired
+    public LessonController(LessonStructureService lessonStructureService) {
+        this.lessonStructureService = lessonStructureService;
+    }
```
гоу инжект через поля  
добавь, кстати в ридмик пометку об этом плз  
это, конечно, холивар, но я так люблю, так тестить удобно  и много чего еще

free2fqn replied:
Ок, поправлю. Начал делать так из-за того что вроде как рекомендовано избегать инжекта через поля

nekromant322 commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureService.java:
```diff
@@ -0,0 +1,77 @@
+
+    @PostConstruct
+    private void postConstruct() {
+        String stringPathToCourses = "platform_orchestrator/src/main/resources/templates/lessons"; /*C:\Users\Schneider\IdeaProjects\platform\platform_orchestrator\src\main\resources\templates\lessons\*/
```
це шо

nekromant322 replied:
убери

nekromant322 reviewed (changes requested):
минорные правки и мержу

free2fqn committed:
clean up
Refactored injecting of LessonStructureService
update README.md

kemenchik commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureService.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.stream.Stream;
+
+
+@Slf4j
```
Есть аннотация, но нет логов нигде

kemenchik commented on platform_orchestrator/src/main/java/com/override/controller/LessonController.java:
```diff
@@ -1,16 +1,29 @@
+@RequestMapping("/lessons")
 public class LessonController {
 
+    @Autowired
+    LessonStructureService lessonStructureService;
```
На сколько мне известно, такие места лучше делать приватными

free2fqn replied:
у меня было private final и инжектил через конструктор - @nekromant322 сказал инжектить через поле.

kemenchik replied:
Так private не запрещает через филд инжектить

free2fqn replied:
пофиксил

kemenchik commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureService.java:
```diff
@@ -0,0 +1,78 @@
+@Component
+public class LessonStructureService {
+
+    HashMap<String, JsonObject> courseLessonStrucutre;
+
+    public JsonObject getLessonStructure(String courseName) {
+        return courseLessonStrucutre.get(courseName);
+    }
```
Есть геттер, но поле не приватное

kemenchik commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureService.java:
```diff
@@ -0,0 +1,78 @@
+    }
+
+    @PostConstruct
+    private void postConstruct() {
```
Имя метода не очень, лучше написать для чего он нужен

kemenchik commented on platform_orchestrator/src/main/java/com/override/service/LessonStructureService.java:
```diff
@@ -0,0 +1,78 @@
+        List<String> resultList;
+        Path path = Paths.get(stringPathToDirectory);
+        Stream<Path> fileStream = Files.list(path);
+        resultList = fileStream
+                .map(Path::getFileName)
+                .map(Path::toString)
+                .collect(Collectors.toList());
```
Что-то подсказывает, что где-то должна быть проверка на то, что это папка, мало ли что

free2fqn replied:
Курсы добавляются не от балды, а по правилам оформления, с ограничением глубины. Я так понимаю что файлы там появиться не могут

kemenchik commented on platform_orchestrator/src/main/java/com/override/PlatformOrchestratorApplication.java:
```diff
@@ -4,14 +4,15 @@
 
     public static void main(String[] args) {
-        SpringApplication.run(PlatformOrchestratorApplication.class, args);
+        ApplicationContext context = SpringApplication.run(PlatformOrchestratorApplication.class, args);
```
Это зачем? реально не понимаю

free2fqn replied:
oops, не подчистил

free2fqn committed:
clean up

free2fqn committed:
clean up
